redaxo/src/core/assets/session-timeout.js

```diff
@@ -0,0 +1,201 @@
+/*
+rex.session_keep_alive // refresh rate
+rex.session_duration // Maximale Sessiondauer wenn nicht aktualisiert wurde
+rex.session_max_overall_duration // (maximale komplette dauer, danach wird IMMER geschlossen
+rex.session_start
+rex.time
+rex.session_warning_time // Zeit in Sekunden vor dem Ablauf einer Session, an der eine Warnung ausgegeben wird, dass die Session bald abläuft
+rex.session_logout_url
+rex.session_keep_alive_url
+
+rex.i18n.session_timeout_title
+rex.i18n.session_timeout_message
+rex.i18n.session_timeout_logout_label
+rex.i18n.session_timeout_refresh_label
+*/
+
+if (1 === rex.session_logged_id && rex.session_keep_alive) {
+
+    (function ($) {
+
+            'use strict';
+
+            $.rex_sessionTimeout = function (options) {
+                rex.currentBrowserTime = new Date(); // 02:00h
+                rex.serverTime = new Date(rex.time * 1000); // 05:00h
+                rex.offsetTime = rex.currentBrowserTime.getTime() - rex.serverTime.getTime(); // -3h
+                rex.lastSessionUpdate = new Date(rex.time * 1000);
+                rex.maxKeepAlive = new Date((rex.time + rex.session_keep_alive) * 1000); // 2 Minuten
+                rex.keepAliveExpandtime = 15 * 60; // 15 Minuten in Sekunden
+
+                if (rex.session_warning_time > rex.session_duration) {
+                    rex.session_warning_time = parseInt(rex.session_duration / 2); // Session duration halbiert
+                }
+
+                $.rex_sessionInterval(); // Session halten
+
+            }
+
+            $.rex_sessionInterval = function (options) {
+
+                var rex_sessionCheckInterval = setInterval(function () {
+
+                    rex.currentBrowserTime = new Date();
+                    rex.currentServerTime = new Date(rex.currentBrowserTime.getTime() - rex.offsetTime);
```

Comment:
Auch die ganzen Variablen würde ich als lokale Variablen setzen, und nicht in `rex` abladen. In `rex` sollten nur die Werte sein, die von PHP aus übergeben werden.